static/src/javascripts/projects/common/modules/commercial/track-ad.js

```diff
@@ -0,0 +1,30 @@
+define([
+    'Promise',
+    'common/utils/mediator'
+], function(Promise, mediator) {
+    return trackAd;
+
+    function trackAd(id) {
```

Comment:
Yeah I failed to do that by trying to be too smart in a previous PR, I think this version is fine. The weak link is those events, `dfp-api` should ultimately keep track of the state of each slot—currently they're removed and forgotten as soon as the ad is rendered.